lib/std/Thread.zig

```diff
@@ -76,6 +76,121 @@ pub fn spinLoopHint() void {
+    .linux => 16,
+    .windows => 32,
+    .macos, .ios, .watchos, .tvos => 64,
+    .netbsd => 32,
+    .freebsd => 16,
+    .openbsd => 32,
+    else => 0,
+};
+
+pub fn setName(self: Thread, name: [:0]const u8) !void {
+    if (name.len >= maxNameLen) return error.NameTooLong;
+
+    switch (std.Target.current.os.tag) {
+        .linux => if (use_pthreads) {
+            const err = c.pthread_setname_np(self.data.handle, name.ptr);
+            return switch (err) {
+                0 => {},
+                else => unreachable,
+            };
+        } else if (self.data.handle == getCurrentId()) {
+            _ = try os.prctl(.SET_NAME, .{@ptrToInt(name.ptr)});
+        } else if (!use_pthreads) {
+            var buf: [32]u8 = undefined;
+            const path = try std.fmt.bufPrint(&buf, "/proc/self/task/{d}/comm", .{self.data.handle});
+
+            const file = try std.fs.cwd().openFile(path, .{ .write = true });
+            defer file.close();
+
+            try file.writer().print("{s}", .{name});
+        },
+        .windows => {
+            // TODO
+        },
+        .macos, .ios, .watchos, .tvos => if (use_pthreads) {
+            // There doesn't seem to be a way to set the name for an arbitrary thread, only the current one.
+            if (self.data.handle != getCurrentId()) return;
+
+            const err = c.pthread_setname_np(name.ptr);
+            return switch (err) {
+                0 => {},
+                else => unreachable,
+            };
+        },
+        .netbsd => if (use_pthreads) {
+            const err = c.pthread_setname_np(self.data.handle, name.ptr, null);
+            return switch (err) {
+                0 => {},
+                else => unreachable,
+            };
+        },
+        .freebsd, .openbsd => if (use_pthreads) {
+            // Use pthread_set_name_np for FreeBSD because the alias pthread_setname_np is FreeBSD 12.2+ only.
+            const err = c.pthread_set_name_np(self.data.handle, name.ptr);
+            return switch (err) {
+                0 => {},
+                else => unreachable,
+            };
+        },
+        else => {},
+    }
+}
+
+pub fn getName(self: Thread, allocator: *std.mem.Allocator) !?[:0]const u8 {
+    // musl doesn't provide pthread_getname_np and there's no way to retrieve the thread id of an arbitrary thread.
+    if (use_pthreads and self.data.handle != getCurrentId() and comptime std.Target.current.abi.isMusl()) {
+        return null;
+    }
+
+    var buffer = try allocator.allocWithOptions(u8, maxNameLen - 1, null, 0);
+    errdefer allocator.free(buffer);
+    std.mem.set(u8, buffer, 0);
+
+    switch (std.Target.current.os.tag) {
+        .linux => if (use_pthreads and comptime std.Target.current.abi.isGnu()) {
+            const err = c.pthread_getname_np(self.data.handle, buffer.ptr, maxNameLen);
+            return switch (err) {
+                0 => std.mem.spanZ(buffer),
+                os.ERANGE => error.NameTooLong,
```

Comment:
Given the hardcoded upper limit on the length is `maxNameLen` this case can be marked as `unreachable`, no?